libafl/src/mutators/numeric.rs

```diff
@@ -148,16 +147,24 @@ impl Named for NegMutator {
 #[derive(Debug)]
 pub struct RandMutator;
 
-impl<I, S> Mutator<I, S> for RandMutator
+impl<I, S> Mutator<ValueInput<I>, S> for RandMutator
 where
     S: HasRand,
-    I: Shl<u32, Output = I> + NumericConsts + BitOrAssign + From<u8> + Zero,
+    ValueInput<I>: Shl<usize, Output = ValueInput<I>> + BitOrAssign,
+    I: From<u8> + Zero,
 {
-    fn mutate(&mut self, state: &mut S, input: &mut I) -> Result<MutationResult, Error> {
-        // set to byte-wise since the RNGs don't work for all numeric types
-        *input = I::zero();
-        for offset in 0..I::BITS {
-            let mask: I = (state.rand_mut().next() as u8).into();
+    fn mutate(
+        &mut self,
+        state: &mut S,
+        input: &mut ValueInput<I>,
+    ) -> Result<MutationResult, Error> {
+        // set to random data byte-wise since the RNGs don't work for all numeric types
+        *input = I::zero().into();
+
+        for offset in 0..(size_of::<I>() % size_of::<u8>()) {
+            let raw = state.rand_mut().next() as u8;
```

Comment:
You should really get one `next()` and use parts of that.
it's 64 bit so you'll need one extra clause for 128 bit vals

Reply:
(but calling rand is a lot slwer than one extra check)

Reply:
yes..
Mutators has to be __fast__

Reply:
Should be better now